BUILDING.md

```diff
@@ -0,0 +1,87 @@
+# Build-Steps
+
+The J2V8 build-system performs several build steps in a fixed order to produce the final J2V8 packages for usage on the designated target platforms. What follows is a short summary for what each of the executed build-steps does and what output artifacts are produced by each step.
+
+---
+## Node.js
+
+Builds the [Node.js](https://nodejs.org/en/) & [V8](https://developers.google.com/v8/) dependency artifacts that are later linked against by the J2V8 native bridge code.
+
+__Inputs:__
+- Node.js source code
+    - see [Github](https://github.com/nodejs/node)
+- Node.js GIT patches with customizations for integrating Node.js into J2V8
+    - `./node.patches/*.diff`
+
+__Artifacts:__
+- Node.js & V8 static link libraries
+    - `./node/out/`
+    - `./node/build/`
+    - `./node/Debug/`
+    - `./node/Release/`
+---
+## CMake
+
+Uses [CMake](https://cmake.org/) to generate the native Makefiles / IDE project files to later build the J2V8 C++ native bridge shared libraries (.so/.dylib/.dll)
+
+__Inputs__:
+- Node.js / V8 static link libraries
+    - `./cmake/NodeJsUtils.cmake`
+- CMakeLists & CMake utilities
+    - `CMakeLists.txt`
+    - `./cmake/*.cmake`
+
+__Artifacts:__
+- CMake generated Makefiles / IDE Project-files
```

Comment:
Thanks .:heart:
I wanted to add more but my time was limited. I will continue documenting what I think would be helpful once I have some more spare time.